### PR TITLE
Fix letter branding preview when no service present

### PR DESCRIPTION
**Pull request by quis**
Because we serve letter branding images without cookies there’s no`current_service` available in the request context.

This was causing an error when trying to preview a letter with no branding, because the fallback is to look at the branding for the current service.

This commit adds a conditional for this case, and makes the test more end-to-end to ensure this edge case is accounted for.

### CODE REVIEW

File: tests/app/main/views/test_templates.py [hunk 1]
```diff
@@ -1937,23 +1937,31 @@ def test_letter_branding_preview_image(
...
         filename=original_filename,
     )
 
     mocked_preview.assert_called_with(
-        {
-            "subject": "An example letter",
-            "content": ANY,
-            "template_type": "letter",
-            "is_precompiled_letter": False,
+        "http://localhost:9999/preview.png",
```
Comment (CrystalPea) [inline tests/app/main/views/test_templates.py:1952]: I don't understand how the assertion worked both with the old version of args and the new version? Not that much changed in the code to change the call so much?

Comment (CrystalPea) [reply tests/app/main/views/test_templates.py:1952]: Ohh, yes, of course, I missed that the mock changed, thank you 🙏🏼

**Review by CrystalPea — APPROVED** [review]
